test/xspec-xmlns.xspec

```diff
@@ -0,0 +1,460 @@
+					test="local-name($Q{http://example.org/ns/my/inherited-space}localvar)" select="'inherited-localvar-name-in-other-space'"/>
+				<x:expect label="and variable value"
+					test="namespace-uri($Q{http://example.org/ns/my/inherited-space}localvar) = xs:anyURI('http://example.org/ns/my/inherited-space')"/>
+			</x:scenario>
+		</x:scenario>
+		<x:scenario label="relying on default element namespace">
+			<x:scenario label="declared on x:description">
+				<x:variable name="localvar-default-element-ns" as="element()"><element/></x:variable>
+				<x:call function="true"/>
+				<x:expect label="correctly uses default namespace of x:description"
+					test="namespace-uri($Q{}localvar-default-element-ns)" select="xs:anyURI('http://docbook.org/ns/docbook')"/>
+			</x:scenario>
+			<x:scenario label="declared on x:scenario" xmlns="http://example.org/ns/my/ns2">
+				<x:variable name="tv1:localvar-element-ns" as="element()"><element/></x:variable>
+				<x:call function="true"/>
+				<x:expect label="correctly uses default namespace of x:scenario"
+					test="namespace-uri($Q{http://example.org/ns/my/variable1}localvar-element-ns)" select="xs:anyURI('http://example.org/ns/my/ns2')"/>
+			</x:scenario>
+			<x:scenario label="declared on x:variable">
+				<x:variable name="tv1:localvar-element-ns"
+					xmlns="http://example.org/ns/my/ns2" as="element()"><element/></x:variable>
+				<x:call function="true"/>
+				<x:expect label="correctly uses default namespace of x:variable"
+					test="namespace-uri($Q{http://example.org/ns/my/variable1}localvar-element-ns)" select="xs:anyURI('http://example.org/ns/my/ns2')"/>
+			</x:scenario>
+		</x:scenario>
+	</x:scenario>
+
+	 <x:scenario label="Assertion">
+		<x:scenario label="relying on xmlns:... on x:expect">
+			<x:call function="exactly-one">
+				<x:param><my1:element xmlns:my1="http://example.org/ns/my/expect"/></x:param>
+			</x:call>
+			<x:expect label="correctly uses local my2 binding for user-content." xmlns:my2="http://example.org/ns/my/expect">
+				<my2:element/>
+			</x:expect>
+			<x:variable name="tv1:expected-element-local-name" select="'globalvar'"/>
+			<x:expect label="correctly uses local tv2 binding to find variable and xsd binding to evaluate expression."
+				test="local-name($tv2:globalvar) treat as xsd:string" xmlns:tv2="http://example.org/ns/my/variable1"
+				xmlns:xsd="http://www.w3.org/2001/XMLSchema" select="$tv2:expected-element-local-name"/>
+			 <x:pending label="Due to issue #639, you cannot reuse prefixes xs or xsl">
+				<x:expect label="even if reusing 'xs' prefix?"
+					test="local-name($xs:globalvar) treat as xsd:string" xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+					xmlns:xs="http://example.org/ns/my/variable1" select="'globalvar'"/>
+				<!-- Locally reusing x or xsl prefix causes error, even if inside x:pending, so comment out.
+					<t:expect label="even if reusing 'x' prefix? (Is this different from #639?)" xmlns:t="http://www.jenitennison.com/xslt/xspec"
```

Comment:
Indeed, this is different from #639 and this pull request should probably handle local reuse of the 'x' prefix. I think I can fix this by making `xspec-name` context-dependent instead of a global variable. I'll upload that change as soon as I have more time to qualify it (though the week ahead looks busy).

Reply:
Agreed that `x:xspec-name()` should be context aware.

Reply:
Done, with commit 0b34eb3190493f3da9e29d459454396f74a86d82